hudi-cli/src/main/java/org/apache/hudi/cli/commands/CommitsCommand.java

```diff
@@ -431,4 +442,20 @@ public String syncCommits(@CliOption(key = {"path"}, help = "Path of the table t
     return "Load sync state between " + HoodieCLI.getTableMetaClient().getTableConfig().getTableName() + " and "
         + HoodieCLI.syncTableMetadata.getTableConfig().getTableName();
   }
+
+  /*
+  Checks whether a commit or replacecommit action exists in the timeline.
+  * */
+  private Option<HoodieInstant> getCommitOrReplaceCommitInstant(HoodieTimeline timeline, String instantTime) {
```

Comment:
consider changing signature to return Option<HoodieCommitMetadata> and deserialize instant details inside this method. This would avoid repetition to get instant details in multiple places. You can also do additional validation. for example: for replace commit, deserialize  using HoodieReplaceCommitMetadata class